config/main.py

```diff
@@ -570,6 +570,102 @@ def is_ipaddress(val):
         return False
     return True
 
+def  interface_is_in_vlan(vlan_member_table, interface_name):
+    """ Check if an interface  is in a vlan """
+    for _,v in vlan_member_table.keys():
+        if v == interface_name:
+            return True
+
+    return False
+
+def  interface_is_in_portchannel(portchannel_member_table, interface_name):
+    """ Check if an interface is part of portchannel """
+    for _,v in portchannel_member_table.keys():
+        if v == interface_name:
+            return True
+
+    return False
+
+def interface_is_router_port(interface_table, interface_name):
+    """ Check if an interface has router config """
+    for entry in interface_table.keys():
```

Comment:
I would also pick a more descriptive name than "entry" as I am not sure what `entry[0]` is supposed to signify. It might be hard for someone to debug later.

Reply:
modified as intf.